src/audio/rtnr/rtnr.c

```diff
@@ -0,0 +1,546 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Realtek Semiconductor Corp. All rights reserved.
+//
+// Author: Ming Jen Tai <mingjen_tai@realtek.com>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/rtnr/rtnr.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <sof/audio/rtnr/rtklib/include/RTK_MA_API.h>
+
+
+#define SAMPLING_RATE		16000
+#define PROCESS_RATE		16000
+#define PRESET_SIZE			1040
+#define MicNum				2
+#define SpkNum				2
+
+#if SAMPLING_RATE == 48000
+#define BLOCKLENGTH	768
+#elif SAMPLING_RATE == 16000
+#define BLOCKLENGTH	256
+#elif SAMPLING_RATE == 8000
+#define BLOCKLENGTH	128
+#endif // SAMPLING_RATE
+
+
+#define RTNR_BLK_LENGTH 4
+#define RTNR_MAX_SOURCES 1 /* Microphone stream */
+
+/*	MIC input is at index 0 */
+#define RTNR_MIC_SOURCE_INDEX (0)
+
+static const struct comp_driver comp_rtnr;
+
+/** \brief RTNR processing functions map item. */
+struct rtnr_func_map {
+	enum sof_ipc_frame fmt; /**< source frame format */
+	rtnr_func func; /**< processing function */
+};
+
+#if CONFIG_FORMAT_S16LE
+static void rtnr_s16_default(struct comp_dev *dev, const struct audio_stream **sources,
+			    struct audio_stream *sink, int frames);
+#endif /* CONFIG_FORMAT_S16LE */
+
+#if CONFIG_FORMAT_S24LE
+static void rtnr_s24_default(struct comp_dev *dev, const struct audio_stream **sources,
+			    struct audio_stream *sink, int frames);
+#endif /* CONFIG_FORMAT_S24LE */
+
+#if CONFIG_FORMAT_S32LE
+static void rtnr_s32_default(struct comp_dev *dev, const struct audio_stream **sources,
+			    struct audio_stream *sink, int frames);
+#endif /* CONFIG_FORMAT_S32LE */
+
+/* Processing functions table */
+const struct rtnr_func_map rtnr_fnmap[] = {
+#if CONFIG_FORMAT_S16LE
+	{ SOF_IPC_FRAME_S16_LE, rtnr_s16_default },
+#endif // CONFIG_FORMAT_S16LE
+#if CONFIG_FORMAT_S24LE
+	{ SOF_IPC_FRAME_S24_4LE, rtnr_s24_default },
+#endif // CONFIG_FORMAT_S24LE
+#if CONFIG_FORMAT_S32LE
+	{ SOF_IPC_FRAME_S32_LE, rtnr_s32_default },
+#endif// CONFIG_FORMAT_S32LE
+};
+
+const size_t rtnr_fncount = ARRAY_SIZE(rtnr_fnmap);
+
+/* UUID 5c7ca334-e15d-11eb-ba80-0242ac130004 */
+DECLARE_SOF_RT_UUID("rtnr", rtnr_uuid, 0x5c7ca334, 0xe15d, 0x11eb, 0xba, 0x80,
+		    0x02, 0x42, 0xac, 0x13, 0x00, 0x04);
+
+DECLARE_TR_CTX(rtnr_tr, SOF_UUID(rtnr_uuid), LOG_LEVEL_INFO);
+
+/* Generic processing */
+
+
+void myPrintf(int a, int b, int c, int d, int e)
```

Comment:
redundant code?

Reply:
This function will be called from the library for debugging purpose.

Reply:
It's ok then please add a comment on the header about this.

Reply:
This has been fixed, thanks.